modules/alloy/src/main/ts/ephox/alloy/menu/util/TieredMenuEvents.ts

```diff
@@ -0,0 +1,30 @@
+import { Id } from '@ephox/katamari';
+
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import { CustomEvent } from '../../events/SimulatedEvent';
+
+// The primary purpose of these events is to bubble up to the tiered menu,
+// and allow the tiered menu handlers to have a reference to
+// all of the tiered menu, the active menu, and the item comp
+// when firing their handlers (like onHighlightItem)
+export interface OnMenuItemHighlightedEventData {
+  menuComp: AlloyComponent;
```

Comment:
Do we allow mutation on event data or should we have readonly annotations on these?

Reply:
- [ ] readonly

Reply:
Updated: added `readonly`